File: 55-jump-game/55-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int reach = 0;
        for(int i =0; i< nums.length; i++){
            if( nums[i]+i > reach){
                reach = nums[i]+i;
            }
            if( reach ==i){
                break;
            }
        }
        return reach>=nums.length-1;
    }
}

File: 55-jump-game/55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int reach = 0;
        for(int i =0; i< nums.size(); i++){
            reach = max(i+nums[i] ,reach);
            if( reach ==i){
                break;
            }
        }
        return reach>=nums.size()-1;
    }
};